io/zenoh-link-commons/src/tls.rs

```diff
@@ -86,3 +86,104 @@ impl WebPkiVerifierAnyServerName {
         Self { roots }
     }
 }
+
+pub mod expiration {
+    use std::{net::SocketAddr, sync::Weak};
+
+    use time::OffsetDateTime;
+    use tokio::task::JoinHandle;
+    use tokio_util::sync::CancellationToken;
+
+    use crate::LinkUnicastTrait;
+
+    #[derive(Debug)]
+    pub struct LinkCertExpirationManager {
+        token: CancellationToken,
+        handle: Option<JoinHandle<()>>,
+    }
+
+    impl LinkCertExpirationManager {
+        pub fn new(
+            link: Weak<dyn LinkUnicastTrait>,
+            src_addr: SocketAddr,
+            dst_addr: SocketAddr,
+            link_type: String,
+            expiration_time: OffsetDateTime,
+        ) -> Self {
+            let token = CancellationToken::new();
+            let handle = zenoh_runtime::ZRuntime::Acceptor.spawn(expiration_task(
+                link,
+                src_addr,
+                dst_addr,
+                link_type,
+                expiration_time,
+                token.clone(),
+            ));
+            Self {
+                token,
+                handle: Some(handle),
+            }
+        }
+    }
+
+    // Cleanup expiration task when manager is dropped
+    impl Drop for LinkCertExpirationManager {
+        fn drop(&mut self) {
+            if let Some(handle) = self.handle.take() {
+                self.token.cancel();
+                zenoh_runtime::ZRuntime::Acceptor.block_in_place(async {
+                    let _ = handle.await;
+                })
+            }
+        }
+    }
+
+    async fn expiration_task(
+        link: Weak<dyn LinkUnicastTrait>,
+        src_addr: SocketAddr,
+        dst_addr: SocketAddr,
+        link_type: String,
```

Comment:
If we need a `String` later, then we can change the type. For now, `&'static str` is enough.